server/lib/pbench/process_tb.py

```diff
@@ -49,25 +50,30 @@ def _get_receive_dir(self) -> Path:
         return receive_dir
 
     @staticmethod
-    def _results_push(controller: str, tb: Path, token: str):
-        """Runs Agent's `pbench-results-push` command with controller, tb
-        and token options
+    def _results_push(tb: Path, token: str, satellite: Optional[str] = None):
+        """Runs Agent's `pbench-results-push` command with tb, token and
+        metadata options
 
         Args -
-            controller -- the name of the controller to be associated with the
-                          tar ball
             tb -- path of the tar ball
             token -- generated authorised token for Pbench user
+            satellite -- prefix of the pbench's satellite server
 
         Importance of this function is while running tests we get the
         ability to mock this function and test it easily.
         """
+        results_push = f"pbench-results-push {tb} --token={token}"
+
+        if satellite:
+            satellite = "server.origin:" + satellite
+            results_push += f" --metadata={satellite}"
```

Comment:
Hmm; I like my previous suggestion of removing the local variable in this `if` even more now that you're modifying the actual parameter value to the method. That is,

```
    if satellite:
        results_push += f" --metadata=server.origin:{satellite}"
```